### PR TITLE
[opt](Nereids) add boolean type signature for sum aggregate function

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
add boolean type for sum agg

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum.java [hunk 1]
```diff
@@ -78,8 +80,9 @@ public Sum(boolean distinct, boolean alwaysNullable, Expression arg) {
     @Override
     public void checkLegalityBeforeTypeCoercion() {
         DataType argType = child().getDataType();
-        if (((!argType.isNumericType() && !argType.isNullType()) || argType.isOnlyMetricType())) {
-            throw new AnalysisException("sum requires a numeric parameter: " + this.toSql());
+        if ((!argType.isNumericType() && !argType.isBooleanType() && !argType.isNullType())
+                || argType.isOnlyMetricType()) {
+            throw new AnalysisException("sum requires a numeric or boolean parameter: " + this.toSql());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum.java:85]: add regression test in `regression-test/suites/nereids_function_p0/agg_function/agg.groovy`

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.42 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17170450849 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum.java (hunk 1 above)
Comment (zhangstar333) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum.java:85]: done

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.98 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162800449 Bytes

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**